### PR TITLE
Remove catch(...) code style

### PR DESCRIPTION
**Pull request by panhaiqi**
### Details:
 - *Remove catch(...) code style*

### Tickets:
 - *96402*

### CODE REVIEW

**Comment by riverlijunjie** [comment]
@panhaiqi  please resolve clang issue.

File: src/plugins/auto/auto_schedule.cpp [hunk 1]
```diff
@@ -310,7 +310,7 @@ void AutoSchedule::TryToLoadNetWork(AutoLoadContext& context, const std::string&
             int maxNumThreads = 0;
             try {
                 maxNumThreads = _autoSContext->_core->GetConfig(device, GPU_CONFIG_KEY(MAX_NUM_THREADS)).as<int>();
-            } catch (...) {
+            } catch (std::exception&) {
```
Comment (ilyachur) [inline src/plugins/auto/auto_schedule.cpp:313]: Can we get any other exceptions expect `ov::Exception` or `InferenceEngine::Exception`?

File: src/bindings/c/src/common.h [hunk 2]
```diff
@@ -34,7 +34,7 @@
     CATCH_OV_EXCEPTION(INFER_NOT_STARTED, InferNotStarted)    \
     CATCH_OV_EXCEPTION(NETWORK_NOT_READ, NetworkNotRead)      \
     CATCH_OV_EXCEPTION(INFER_CANCELLED, InferCancelled)       \
-    catch (...) {                                             \
+    catch (std::exception&) {                                 \
```
Comment (alalek) [inline src/bindings/c/src/common.h:37]: "const reference"

Ref: https://en.cppreference.com/w/cpp/language/try_catch

File: src/bindings/c/src/ie_c_api.cpp [hunk 3]
```diff
@@ -189,7 +189,7 @@ std::map<IE::ColorFormat, colorformat_e> colorformat_map = {{IE::ColorFormat::RA
     CATCH_IE_EXCEPTION(INFER_NOT_STARTED, InferNotStarted)    \
     CATCH_IE_EXCEPTION(NETWORK_NOT_READ, NetworkNotRead)      \
     CATCH_IE_EXCEPTION(INFER_CANCELLED, InferCancelled)       \
-    catch (...) {                                             \
+    catch (const std::exception&) {                                 \
```
Comment (riverlijunjie) [inline src/bindings/c/src/ie_c_api.cpp:192]: clang format issue - to align the place of "\"

File: src/plugins/auto/auto_executable_network.cpp [hunk 4]
```diff
@@ -92,7 +92,7 @@ IE::Parameter AutoExecutableNetwork::GetMetric(const std::string& name) const {
                 bThroughputEnabledInPlugin =
                     _autoSContext->_core->GetConfig(deviceInfo.deviceName,
                         CONFIG_KEY(PERFORMANCE_HINT)).as<std::string>() == CONFIG_VALUE(THROUGHPUT);
-            } catch (...) {
+            } catch (const IE::Exception&) {
```
Comment (riverlijunjie) [inline src/plugins/auto/auto_executable_network.cpp:95]: IE::Exception is deprecated class, why not use ov::Exception?

Comment (riverlijunjie) [reply src/plugins/auto/auto_executable_network.cpp]: For currently, plugin still use legacy IE::Exception, but for api level it is better to ov::Exception.
Sorry to mislead.

**Review by riverlijunjie — APPROVED** [review]
LGTM

**Review by ilyachur — APPROVED** [review]

**Comment by ilyachur** [comment]
Please fix CI

File: src/plugins/auto/auto_schedule.cpp [hunk 5]
```diff
@@ -323,7 +323,7 @@ void AutoSchedule::TryToLoadNetWork(AutoLoadContext& context, const std::string&
             int maxNumThreads = 0;
             try {
                 maxNumThreads = _autoSContext->_core->GetConfig(device, GPU_CONFIG_KEY(MAX_NUM_THREADS)).as<int>();
-            } catch (...) {
+            } catch (const InferenceEngine::Exception&) {
```
Comment (songbell) [inline src/plugins/auto/auto_schedule.cpp:326]: InferenceEngine->IE

**Comment by wgzintel** [comment]
On hold for waiting another PR to fix core dump.

**Review by songbell — APPROVED** [review]

**Comment by peterchen-intel** [comment]
GTM

**Review by ilyachur — APPROVED** [review]

**Review by riverlijunjie — APPROVED** [review]

**Review by wangleis — APPROVED** [review]

**Pull request merged**